examples/plugin/com_osvr_example_Tracker.cpp

```diff
@@ -0,0 +1,122 @@
+/** @date 2015
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/PluginKit/PluginKit.h>
+#include <osvr/PluginKit/TrackerInterfaceC.h>
+
+// Generated JSON header file
+#include "com_osvr_example_Tracker_json.h"
+
+// Library/third-party includes
+
+// Standard includes
+#include <iostream>
+#include <chrono>
+#include <thread>
+#include <random>
+#include <ctime>
+
+// Anonymous namespace to avoid symbol collision
+namespace {
+
+class TrackerDevice {
+  public:
+    TrackerDevice(OSVR_PluginRegContext ctx) {
+        /// Create the initialization options
+        OSVR_DeviceInitOptions opts = osvrDeviceCreateInitOptions(ctx);
+
+        // configure our tracker
+        osvrDeviceTrackerConfigure(opts, &m_tracker);
+
+        /// Create the device token with the options
+        m_dev.initAsync(ctx, "Tracker", opts);
+
+        /// Send JSON descriptor
+        m_dev.sendJsonDescriptor(com_osvr_example_Tracker_json);
+
+        /// Register update callback
+        m_dev.registerUpdateCallback(this);
+
+        /// Seed pseudo random generator.
+        std::srand(std::time(0));
+    }
+
+    OSVR_ReturnCode update() {
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(
+            5)); // Simulate waiting a quarter second for data.
+
+        OSVR_TimeValue now;
+        osvrTimeValueGetNow(&now);
+        double t = 5.0 * ((double)now.seconds + ((double)now.microseconds / 1000000.0));
```

Comment:
Time since epoch vs time since start doesn't matter - the values I'm sending for position are cyclical. The point was just to get some kind of movement in there so you can tell it's working. t still represents time, just in some unnamed unit defined as 5 units per second.